corehq/apps/geospatial/models.py

```diff
@@ -81,3 +83,22 @@ def disbursement_solver(self):
         return self.VALID_DISBURSEMENT_ALGORITHM_CLASSES[
             self.selected_disbursement_algorithm
         ]
+
+    @property
+    def plaintext_api_token(self):
+        if self.api_token and self.api_token.startswith(f'${ALGO_AES}$'):
+            ciphertext = self.api_token.split('$', 2)[2]
+            return b64_aes_decrypt(ciphertext)
+        return self.api_token
+
+    @plaintext_api_token.setter
+    def plaintext_api_token(self, value):
+        if value is None:
+            self.api_token = None
+        elif not isinstance(value, str):
+            raise Exception("Only string values allowed for plaintext api token")
```

Comment:
How is this propegated to the user? Do we catch it and show a message to the user? If so, we should translate it as well

Reply:
This isn't being propagated, as the input from the user is always string. I just thought I'd add this as an additional safeguard.

Reply:
In the very unlikely event that this exception gets thrown, would it then result in an uncaught exception? If so, I wonder if it might be better to at least safely catch this from where we're calling this?

Reply:
I do agree it might be prudent to catch the exception, but I'm not 100% sure what to do with it, unless I do something like the following...
```
    def save(self, commit=True):
        if self.cleaned_data.get('plaintext_api_token'):
            try:
                self.instance.plaintext_api_token = self.cleaned_data.get('plaintext_api_token')
            except ValueError:
                pass
        return super().save(commit)
```
This will simply not save the erroneous token, but without telling the user why.

The form already handles the case where a token is an empty string, but not ints or bools...since the user can't set a faulty value from the UI, I don't think there's much sense in trying to report such a case to the user.  

Thoughts?

Reply:
I see the form specifies this field to be a CharField, so all characters will be interpreted as string already, making this check redundant?

Reply:
Instead of doing a type check here, what if we handled this in the `clean()` function of the form? We can check there whether the type is a string and, if not, raise an appropriate `ValidationError`. This way, we can be sure that the data coming from the front-end is a string and return an appropriate error if not.